dependencies/squid-proxy/certificate.yaml

```diff
@@ -0,0 +1,25 @@
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: squid-proxy-cert
+  namespace: squid-proxy
+  labels:
+    app.kubernetes.io/name: squid-proxy
+    app.kubernetes.io/part-of: konflux
+spec:
+  dnsNames:
+  - squid-proxy.squid-proxy.svc
+  - squid-proxy.squid-proxy.svc.cluster.local
+  - squid-proxy.local
+  - localhost
+  isCA: true  
+  subject:
+    organizations:
+      - konflux
+    organizationalUnits:
+      - Squid Proxy
+  issuerRef:
+    kind: ClusterIssuer
+    name: ca-issuer
```

Comment:
I think this effectively make this CA a subordinate CA to the existing cluster-level "ca-issuer" I'm not sure we want that

Reply:
self-signed-cluster-issuer -> creates root-secret (cluster CA)
ca-issuer -> uses root-secret to sign other certificates
root-secret -> is already in the trust bundle

Do you think we need to update the issuerRef to [self-signed-cluster-issuer](https://github.com/konflux-ci/konflux-ci/blob/main/dependencies/cluster-issuer/self-signed-cluster-issuer.yaml)?
Note that if we will, we will have to update the [bundle](https://github.com/konflux-ci/konflux-ci/blob/main/dependencies/registry/trust-bundle.yaml) to add the secret that will be created by squid-proxy.

Reply:
I want to have separate CAs for different things - it lowers the dangers posed of leaking the private keys - this does mean we will need to work harder to update the trust bundles.

Lets try not to use cluster issuers unless we find reasons why we have to. In the case of Squid I expect all the certificate signings will happen in the "proxy" namespace, so it seems to me like we may not need cluster issuers at all.